src/cartesian/Area.tsx

```diff
@@ -256,7 +256,8 @@ export class Area extends PureComponent<Props, State> {
       dotItem = option(props);
     } else {
       const className = clsx('recharts-area-dot', typeof option !== 'boolean' ? option.className : '');
-      dotItem = <Dot {...props} className={className} />;
+      const { key, ...rest } = props;
+      dotItem = <Dot key={key} {...rest} className={className} />;
```

Comment:
Spreading `key` is deprecated. `<div {...props} key={key} />` would be an alternative to avoid the warning but does deopt to `createElement` instead of the new, fast JSX runtime.